File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int l=0;
        int r=1;
        int mpro=0;
        while(r<prices.length){
            if(prices[l]<prices[r]){
                int p=prices[r]-prices[l];
                mpro=Math.max(mpro,p);

            }
            else{
                l=r;
            }
            r++;
        
        }
        return mpro;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        //2ptr conc
        int l = 0;
        int r = 1;
        int maxpro = 0;

        while (r < prices.size()) {
            if (prices[l] < prices[r]) {
                int pro = prices[r] - prices[l];
                maxpro = max(pro, maxpro);
            } else {
                l = r;
            }
            r += 1;
        }
        return maxpro;
    }
};